test/framework/utils/utils.go

```diff
@@ -11,16 +12,19 @@ func ShellEscape(arg string) string {
 }
 
 func GetFreePort() (int, error) {
-	address, err := net.ResolveTCPAddr("tcp", "localhost:0")
-	if err != nil {
-		return 0, err
```

Comment:
Didn't like the old way? I feel like I changed from random due to collisions but I could be wrong...

Reply:
This exists in pkg/test/network.go btw

Reply:
I had troubles with random ports, I had doubts that it works well with multiple processes.
I offloaded randomizing port for docker for the majority of cases so let me revert this change for now.